providers/writing_a_new_provider.md

```diff
@@ -0,0 +1,555 @@
+create  app/models/manageiq/providers/awesome_cloud/inventory/parser/cloud_manager.rb
+create  app/models/manageiq/providers/awesome_cloud/inventory/persister/definitions/cloud_collections.rb
+create  app/models/manageiq/providers/awesome_cloud/inventory/persister/cloud_manager.rb
+identical  app/models/manageiq/providers/awesome_cloud/inventory/persister.rb
+create  app/models/manageiq/providers/awesome_cloud/cloud_manager.rb
+insert  .yamllint
+append  spec/spec_helper.rb
+```
+
+That's a lot of stuff!  We'll cover all of it in detail later but for now lets take a look at our plugin:
+
+```bash
+$ ls plugins/
+manageiq-providers-awesome_cloud
+```
+
+The generator also automatically adds your plugin to your local Gemfile:
+
+```bash
+$ git diff
+diff --git a/Gemfile b/Gemfile
+index b7e6783821..685474c570 100644
+--- a/Gemfile
++++ b/Gemfile
+ # This default is used to automatically require all of our gems in processes that don't specify which bundler groups they want.
+ #
+ ### providers
++
++group :awesome_cloud, :manageiq_default do
++  manageiq_plugin "manageiq-providers-awesome_cloud"
++end
++
+```
+
+This is a bit optimistic since this hasn't been accepted into the ManageIQ organization yet. :)
+
+To work on this plugin locally you have to tell bundler to look in a different place for your gem (more info in [developer_setup/plugins.md](../developer_setup/plugins.md))
+
+```bash
+$ echo 'override_gem "manageiq-providers-awesome_cloud", :path => "../plugins/manageiq-providers-awesome_cloud"' >> bundler.d/override.rb
+$ bundle update
+```
+
+This tells your core repo where to find your local changes, now lets let your plugin know where your local core repo is:
+```bash
+$ ln -s $(pwd) plugins/manageiq-providers-awesome_cloud/spec/manageiq
+$ cd plugins/manageiq-providers-awesome_cloud
+$ bin/setup
+```
+
+Lets also take this opportunity to commit the initial code built by the generator before we make any changes:
+```bash
+$ git add .
+$ git commit  -m "Initial commit"
+```
+
+Now that we have both sides linked up lets verify that everything worked:
+```ruby
+$ bundle exec rails c
+>> ManageIQ::Providers::AwesomeCloud
+=> ManageIQ::Providers::AwesomeCloud
+>> ManageIQ::Providers::AwesomeCloud::CloudManager
+=> ManageIQ::Providers::AwesomeCloud::CloudManager (call 'ManageIQ::Providers::AwesomeCloud::CloudManager.connection' to establish a connection)
+```
+
+Success!  That means that core ManageIQ knows about our new cloud provider.
+
+Now lets get that provider added so we have something to play with:
+```ruby
+>> ems = ManageIQ::Providers::AwesomeCloud::CloudManager.create!(:name => "My Awesome Cloud", :zone => Zone.default_zone)
+```
+
+Now that we have that done it is time to start filling out our new provider.  The first step is to find the SDK gem for this provider.  If there isn't a provider SDK for Ruby you have a few options which we'll cover later.  For now lets assume that Awesome Cloud has a ruby gem called 'awesome_cloud'.
```

Comment:
Not for this PR, but we might want to rename this fake gem to "awesome_cloud_client" or "awesome_cloud_sdk" or something to make it clearer.